diff --git a/node_modules/@walletconnect/legacy-types/dist/cjs/jsonrpc.d.ts b/node_modules/@walletconnect/legacy-types/dist/cjs/jsonrpc.d.ts
index 7c2f8f0..0cca28c 100644
--- a/node_modules/@walletconnect/legacy-types/dist/cjs/jsonrpc.d.ts
+++ b/node_modules/@walletconnect/legacy-types/dist/cjs/jsonrpc.d.ts
@@ -1,4 +1,4 @@
-import { IClientMeta, IConnector } from "./protocol";
+import { IClientMeta, IConnector, IWalletConnectSession } from "./protocol";
 import { IEvents } from "./events";
 import { IQRCodeModal, IQRCodeModalOptions } from "./qrcode";
 export interface IRpcConnection extends IEvents {
@@ -68,6 +68,7 @@ export interface IWCRpcConnectionOptions {
     signingMethods?: string[];
     qrcodeModalOptions?: IQRCodeModalOptions;
     clientMeta?: IClientMeta;
+    session?: IWalletConnectSession;
 }
 export interface IWCEthRpcConnectionOptions extends IWCRpcConnectionOptions {
     rpc?: IRPCMap;
