diff --git a/node_modules/@walletconnect/legacy-provider/dist/cjs/signer.js b/node_modules/@walletconnect/legacy-provider/dist/cjs/signer.js
index 73d732a..d81cdae 100644
--- a/node_modules/@walletconnect/legacy-provider/dist/cjs/signer.js
+++ b/node_modules/@walletconnect/legacy-provider/dist/cjs/signer.js
@@ -89,9 +89,10 @@ class SignerConnection extends jsonrpc_types_1.IJsonRpcConnection {
         this.qrcode = typeof (opts === null || opts === void 0 ? void 0 : opts.qrcode) === "undefined" || opts.qrcode !== false;
         this.chainId = typeof (opts === null || opts === void 0 ? void 0 : opts.chainId) !== "undefined" ? opts.chainId : this.chainId;
         this.qrcodeModalOptions = opts === null || opts === void 0 ? void 0 : opts.qrcodeModalOptions;
+        const qrCodeModal = opts?.qrCodeModal ? opts.qrCodeModal : legacy_modal_1.default;
         const connectorOpts = {
             bridge: this.bridge,
-            qrcodeModal: this.qrcode ? legacy_modal_1.default : undefined,
+            qrcodeModal: this.qrcode ? qrCodeModal : undefined,
             qrcodeModalOptions: this.qrcodeModalOptions,
             storageId: opts === null || opts === void 0 ? void 0 : opts.storageId,
             signingMethods: opts === null || opts === void 0 ? void 0 : opts.signingMethods,
